airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,737 @@
+
+            # Scaling up the timeout based on the number of nodegroups that are being processed.
+            additional_seconds = 5 * 60
+            countdown = TIMEOUT_SECONDS + (nodegroup_count * additional_seconds)
+            while len(eks_hook.list_nodegroups(clusterName=self.clusterName).get('nodegroups')) > 0:
+                if countdown >= CHECK_INTERVAL_SECONDS:
+                    countdown -= CHECK_INTERVAL_SECONDS
+                    sleep(CHECK_INTERVAL_SECONDS)
+                    self.log.info(
+                        "Waiting for the remaining %s nodegroups to delete.  Checking again in %d seconds.",
+                        nodegroup_count,
+                        CHECK_INTERVAL_SECONDS,
+                    )
+                else:
+                    message = "Nodegroups are still inactive after the allocated time limit.  Aborting."
+                    self.log.error(message)
+                    raise RuntimeError(message)
+
+        self.log.info("No nodegroups remain, deleting cluster.")
+        return eks_hook.delete_cluster(name=self.clusterName)
+
+
+class EKSDeleteNodegroupOperator(BaseOperator):
+    """
+    Deletes an Amazon EKS Nodegroup from an Amazon EKS Cluster.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:EKSDeleteNodegroupOperator`
+
+    :param cluster_name: The name of the Amazon EKS Cluster that is associated with your nodegroup.
+    :type cluster_name: str
+    :param nodegroup_name: The name of the nodegroup to delete.
+    :type nodegroup_name: str
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+         If this is None or empty then the default boto3 behaviour is used. If
+         running Airflow in a distributed manner and aws_conn_id is None or
+         empty, then the default boto3 configuration would be used (and must be
+         maintained on each worker node).
+    :type aws_conn_id: str
+
+    """
+
+    def __init__(
+        self,
+        cluster_name: str,
+        nodegroup_name: str,
+        conn_id: Optional[str] = CONN_ID,
+        region: Optional[str] = REGION,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.clusterName = cluster_name
+        self.nodegroupName = nodegroup_name
+        self.conn_id = conn_id
+        self.region = region
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.conn_id,
+            region_name=self.region,
+        )
+
+        return eks_hook.delete_nodegroup(clusterName=self.clusterName, nodegroupName=self.nodegroupName)
+
+
+class EKSDescribeAllClustersOperator(BaseOperator):
+    """
+    Describes all Amazon EKS Clusters in your AWS account.
+
+    :param max_results: The maximum number of results to return.
+    :type max_results: int
+    :param next_token: The nextToken value returned from a previous paginated execution.
```

Comment:
How is the user to pass this parameter?  Airflow does not support dynamic workflows, but the page count is dynamic.

Reply:
I'm not sure I follow the question.   These are passed in from the DAG like any other parameter.

Many of the EKS APIs have a max result value of 100 entries so there should be a way to get the next paginated set.  They could get the nextToken through a prior task's XCOM or whatever, but I felt it should be available.

Reply:
When defining a DAG, the user should define all tasks. Here, however, there is a case where the number of tasks is not known until the first page is downloaded, and therefore cannot be defined in Airflow.

In the case of Google, we've always fetched all items from all pages. This way, the user could access all elements and did not have to define dynamic DAGs.

Reply:
I see.  I implemented the hooks to follow the API as closely as possible.   You think it would be better for the hook to just return everything and not allow pagination at all?

Reply:
Generally the hook should handle the pagination/return a paginator (see the Athena and Glue hooks) and then the operator just iterates over it.

Reply:
I think I understand what you are saying, thanks.  A comment below suggests that the List operators should be dropped.  I will either modify this or drop it, depending on how that discussion goes.  Thanks for the suggestions, I'll keep it mind int he future either way.

See: https://github.com/apache/airflow/pull/16571#discussion_r660444860

Reply:
Looks like they are staying; corrected in coming revision to return entire list.

Reply:
I think this should be corrected in https://github.com/apache/airflow/pull/16571/commits/fa75d5839682136fc80a2578746ecf6ef2f31822